packages/grafana-ui/src/components/Forms/FieldArray.story.tsx

```diff
@@ -30,8 +30,16 @@ export const simple = () => {
                 <div style={{ marginBottom: '1rem' }}>
                   {fields.map((field, index) => (
                     <HorizontalGroup key={field.id}>
-                      <Input ref={register()} name={`people[${index}].firstName`} value={field.firstName} />
-                      <Input ref={register()} name={`people[${index}].lastName`} value={field.lastName} />
+                      <Input
+                        key={field.id}
+                        {...register(`people.${index}.firstName` as const)}
```

Comment:
`as const`? never head of such a cast.  Can you say some more about this?

Reply:
This is necessary to strictly type check `name` and comes from TS type inference: https://devblogs.microsoft.com/typescript/announcing-typescript-4-2/#reverting-template-literal-inference